code/src/components/app-input/AppInput.scss

```diff
@@ -1,12 +1,33 @@
 @import "@design-system";
 
-.spa-app-input {
-  &__textfield {
-    .MuiInputBase-input {
-      padding: spacing(3) spacing(4);
-    }
-    .MuiFormLabel-root {
-      padding-top: 3px;
+$spa-input-colors: (
+  "contained": var(--spa-input-contained-color, $purple),
+  "success": var(--spa-input-success-color, $green),
+  "danger": var(--spa-input-danger-color, $red),
+  "light": var(--spa-input-light-color, $white),
+  "dark": var(--spa-input-dark-color, $black)
+);
+
+.spa-input {
+  .MuiInputBase-input {
+    padding: spacing(3) spacing(4);
+  }
+
+  .MuiFormLabel-root {
+    padding-top: 3px;
```

Comment:
I suggest to add `spacing(1)` like with one approach everywhere. 1 px it's not so required I guess

Reply:
Refactored